plugins/casutg/src/main/java/com/blazemeter/jmeter/threads/AbstractThreadStarter.java

```diff
@@ -44,11 +44,12 @@ public void run() {
             supplyActiveThreads();
         } catch (InterruptedException e) {
             log.debug("Interrupted", e);
+            Thread.currentThread().interrupt();
```

Comment:
Why is this needed? I assume the `run()` method will just finish and corresponding thread will finish.

Reply:
AFAIU :  "InterruptedException" should not be ignored (squid:S2142)

Reply:
Sorry, I don't understand this reason. It is not ignored, it is logged. But the situation does not require any action apart from logging, the thread will just finish its job naturally.

Reply:
This is what the squid:S2142 says: IMU we must interrupt to be correct:

> InterruptedExceptions should never be ignored in the code, and simply logging the exception counts in this case as "ignoring". The throwing of the InterruptedException clears the interrupted state of the Thread, so if the exception is not handled properly the fact that the thread was interrupted will be lost. Instead, InterruptedExceptions should either be rethrown - immediately or after cleaning up the method's state - or the thread should be re-interrupted by calling Thread.interrupt() even if this is supposed to be a single-threaded application. Any other course of action risks delaying thread shutdown and loses the information that the thread was interrupted - probably without finishing its task.

Reply:
Still I don't see how doing interrupt changes anything... The code is so small and simple, it has not finish the thread...
Ok, let's keep it for now.